client/src/app/components/common/save-as-modal.component.ts

```diff
@@ -43,15 +43,25 @@ import {ModalService} from "../modal";
 `
 })
 export class SaveAsModalComponent {
+
+    /** Path of the source file that should be copied */
+    public filePath: string;
+
+    /** Toggles the cog loader */
     private isCreatingFile: boolean;
+
+    /** Holds the error object that should be shown below the form if exists */
     private error: {[message: string]: string};
+
+    /** New file destination entry form */
     private newFileForm: ControlGroup;
 
     constructor(private formBuilder: FormBuilder,
                 private eventHub: EventHubService,
-                private workspace: WorkspaceService,
                 private modal: ModalService) {
 
+        this.filePath = undefined;
```

Comment:
I think that's what @ivanbatic was saying about it technically being the same in most cases (if a property is set to `undefined` or if it is `undefined` because it isn't there). However, he is correct that if you don't set the property value, it isn't enumerable. This behavior I was able to reproduce: https://jsbin.com/mixagixima/1/edit?html,js,console

The first class doesn't set the property to `undefined` and it isn't one of its enumerable properties (gotten from `Object.keys`). The second class does set its property to `undefined` and it is enumerable.